papermill/execute.py

```diff
@@ -169,7 +170,10 @@ def execute_notebook(notebook, output, parameters=None, kernel_name=None, progre
     # Write final Notebook to disk.
     write_ipynb(nb, output)
     raise_for_execution_errors(nb, output)
-
+    if dataframe_file:
+        outputnb = read_notebook(output)
```

Comment:
i'm not sure this is the most elegant way, but `nb.dataframe` didn't seem to work

Reply:
Yeah it looks like it's not easy to get a `papermill.api.Notebook` object straightaway with this.

Looks like you'd need to take this `nb` (nbformat object) and do the following:

```python
from .api import Notebook

n = Notebook() # we should probably let this take an `nbformat` node object
n.path = 'whateverthepathis'
n.node = nb
```

Reply:
Probably in a separate PR, I think we should let `papermill.api.Notebook` accept both the `path` (optional) and the `node` in it's `__init__` to be able to take the `nbformat` node. That would be super helpful for this PR and future ones.

Reply:
something like this https://github.com/nteract/papermill/pull/57 ?

Reply:
Merged, thanks!

Reply:
Coming back to this now I am confused why `nb == outputnb` isn't true. Would be good if we can add a comment because I predict we will be confused by this again :-/